src/operator/pad-inl.h

```diff
@@ -161,7 +170,29 @@ class PadProp : public OperatorProperty {
     CHECK_EQ(in_shape->size(), 1U) << "Can only be one input to symbol.";
 
     const TShape &dshape = (*in_shape)[pad_enum::kData];
-    if (dshape.ndim() == 0) return false;
+
+    auto rank = dshape.ndim();
+    auto pad = param_.pad_width;
+    auto pad_spec_len = param_.pad_width.ndim();
+
+    if (rank == 0) return false;
+    if ((rank != 4) && (rank != 5)) {
+      LOG(FATAL) << "Current implementation only supports 4-D or 5-D input.";
+    }
+    if (pad[0] || pad[1] || pad[2] || pad[3]) {
+      LOG(FATAL) << "Current implementation expects padding on the first two axes to be zero.";
+    }
+    if ((2*rank) != pad_spec_len) {
+      LOG(FATAL) << "Input shape vs padding spec mismatch.";
+    }
+    if (param_.mode == pad_enum::kReflect) {
+      auto size = dshape.data();
+      if ((pad[4] >= size[2]) || (pad[5] >= size[2]) ||
```

Comment:
No, it starts when equal. Suppose you want to pad `(a, b, c)` on the right with padding size 2, you would get `(a, b, c, b, a)`. So padding size 3 is the point at which you have to bounce off the edge, getting `(a, b, c, b, a, b)`.